modules/auxiliary/scanner/http/cisco_secure_desktop.rb

```diff
@@ -0,0 +1,337 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/proto/http'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Cisco Secure Deskop Bruteforce Login Utility',
+      'Description'    => %{
+        This module scans for Cisco Secure Desktop web login portals and
+        performs login brute force to identify valid credentials.
+      },
+      'Author'         =>
+        [
+          'jnqpblc <jnqpblc[at]gmail.com>', # Revision Author
+          #'Jonathan Claudius <jclaudius[at]trustwave.com>', # Original PoC Author
+        ],
+      'License'        => MSF_LICENSE,
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'USERNAME' => 'cisco',
+          'PASSWORD' => 'cisco'
+        }
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('GROUP', [false, "A specific VPN group to use", ''])
+      ])
+  end
+
+  def run_host(ip)
+    unless check_conn?
+      vprint_error("Connection failed, Aborting...")
+      return false
+    end
+
+    unless is_app_secure_desktop?
+      vprint_error("Application does not appear to be Cisco Secure Desktop. Module will not continue.")
+      return false
+    end
+
+    vprint_good("Application appears to be Cisco Secure Desktop. Module will continue.")
+
+    $sdesktop_cookie = get_token_cookie()
```

Comment:
Instance variables are the lesser of two evils when compared to global variables.

`@sdesktop_cookie = get_token_cookie`

Reply:
Ok, will research. Ruby n00b here... :D

Reply:
Changed $sdesktop_cookie to @sdesktop_cookie and everything seems to work fine. Will commit this change.